### DISCUSSION TITLE
Saving date with timezone via js-sdk in date field

### DISCUSSION THREAD

**Discussion by wiesson** [supabase/supabase#7549] [Questions]
If I save a date e.g. `2022-06-30T00:00:00+02:00` (which is `2022-07-01T00:00:00` in my timezone) into a date field, it will be saved as `2022-06-30`. I didn't think about it, but I either expect an error here or the timezone should be considered. Did I miss something?

I did like this because it was very convenient to not parse the JS date before saving it via the JS SDK, but now all my dates have an offset of one day 😅

**Comment by burggraf** [comment]
Did you create your field as a `TIMESTAMP` or `TIMESTAMPTZ`?  Because if you use the TZ version, Postgres will automatically convert it to UTC when saving.  [Understanding PostgreSQL Timestamp Data Types](https://www.postgresqltutorial.com/postgresql-tutorial/postgresql-timestamp)

**Reply by wiesson** [reply]
I'm using a simple `date` field

**Reply by yw662** [reply]
How can I preserve timezone info if I use timestampz ?

**Comment by steve-chavez** [comment]
You basically want this to fail right:

```sql
select '2022-06-30T00:00:00+02:00'::date;

    date
------------
 2022-06-30
(1 row)
```

Right now PostreSQL is lenient when parsing a date. There was a similar request on https://github.com/supabase/supabase/issues/4803 (workaround is possible by defining a DOMAN as mentioned there).

**Reply by wiesson** [reply]
In your example, I'd like to see `2022-07-01` but I'm not sure if that is correct. I have a use case, where the time is not very important, just the date and I thought, ok cool - just save it, it will be converted to UTC anyway, but that is not the case. The frontend will now format the date before sending it so my problem is solved, but a error message would have been helpful
